include/envoy/upstream/health_checker.h

```diff
@@ -8,6 +8,10 @@
 namespace Envoy {
 namespace Upstream {
 
+enum class HealthState : uint8_t { unhealthy = 0, healthy = 1 };
```

Comment:
nit: Unless there is a real perf reason I would just do a basic: 
```
enum class HealthState { Unhealthy, Healthy };
```

(Upper case for enum values). Same below.